example/conversion_factor.cpp

```diff
@@ -33,7 +33,7 @@ template<mp_units::Quantity Target, mp_units::Quantity Source>
   requires std::constructible_from<Target, Source>
 inline constexpr double conversion_factor(Target, Source)
 {
-  return mp_units::value_cast<Target::unit>(1. * Source::reference).value();
+  return mp_units::value_cast<Target::unit>(1. * Source::reference).value_ref_in(Target::unit);
```

Comment:
I think I now understand the use case.
You _need_ to use `value_ref_in` to prevent another copy.